bloom-housing/bloom
# Api pass key

## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/4008)


## Description
This creates a new api-pass-key guard that is applied through out the backend

## How Can This Be Tested/Reviewed?
**Add the new api .env variable**: API_PASS_KEY

if you use the site from either the public or partners sites, everything should work as normal

if you attempt to use the `localhost:3101/api` or `localhost:3100/api` sites (depending on your local set up) only 2 endpoints should be usable the `/` and the `/jurisdictions` endpoints

everything else should give you an error

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
